cms/admin/forms.py

```diff
@@ -825,6 +875,8 @@ class Meta:
     def save(self, *args, **kwargs):
         page = super(PagePermissionForm, self).save(*args, **kwargs)
         page.clear_cache(menu=True)
+        if "limit_visibility_in_menu" in self.cleaned_data:
```

Comment:
do this above the page declaration and check changed_data instead of cleaned_data.
also is it intentional to update all titles? instead of filter by language?